lxd/auth/driver_tls.go

```diff
@@ -177,15 +180,23 @@ func (t *tls) GetPermissionChecker(ctx context.Context, r *http.Request, entitle
 			return false
 		}
 
+		// GetPermissionChecker can only be used to check permissions on entities of the same type, e.g. a list of instances.
+		if eType != entityType {
+			logger.Warn("Permission checker received URL with unexpected entity type", logger.Ctx{"expected": entityType, "actual": eType, "entity_url": entityURL})
+			return false
+		}
+
+		// If it's a project URL, the project name is in the path, not the query parameter.
 		if eType == entity.TypeProject {
```

Comment:
The duplication of this logic here makes another argument that it should be in ParseURL.